openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1640,17 +1674,33 @@ static int certificate_cb(SSL* ssl, void* arg) {
         issuers = principalBytes(e, SSL_get_client_CA_list(ssl));
     }
 
-    // Execute the java callback
-    (*e)->CallVoidMethod(e, c->certificate_callback, c->certificate_callback_method,
-             P2J(ssl), types, issuers);
+    // Let's check if we should provide the certificate callback as task that can be run on another Thread.
+    if (c->use_tasks != 0) {
+        // Lets create the CertificateCallbackTask and store it on the SSL object. We then later retrieve it via
+        // SSL.getTask(ssl) and run it.
+        jobject task = (*e)->NewObject(e, certificateCallbackTask_class, certificateCallbackTask_init, P2J(ssl), types, issuers, c->certificate_callback);
```

Comment:
I think you need to check for an exception after this line.

Reply:
the constructor does not throw so I guess the only exception that could happen is a `OOME`.

Reply:
Can you put that as a comment in CertificateCallbackTask's ctor, so it's clear it can't throw in the future too?

Reply:
done